xapian-core/cluster/cluster.cc

```diff
@@ -110,18 +121,16 @@ TermListGroup::add_document(const Document &document) {
 }
 
 void
-TermListGroup::add_documents(DocumentSource &docs) {
+TermListGroup::add_documents(MSetDocumentSource docs) {
```

Comment:
It would be beneficial to provide a documents to the clusterer in different ways through the document source. I can't think of any other way for now, except a vector of Documents. But it would be beneficial later on.
I could just make the parameter accept a DocumentSource rather, so we can pass in subclasses of DocumentSource later on.